src/units/unit.cpp

```diff
@@ -2102,11 +2120,13 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 			std::vector<std::string> temp_overlays = utils::parenthetical_split(add, ',');
 			std::vector<std::string>::iterator it;
 			for(it=temp_overlays.begin();it<temp_overlays.end();++it) {
```

Comment:
It seems both this and the similar code on line 1916 could use the `for(const auto& x : temp_overlays)` syntax.